frontend/src/screens/ConnectPartner.tsx

```diff
@@ -8,28 +8,88 @@ import {
   View,
 } from 'react-native';
 import React, { useState } from 'react';
+import { useRecoilValue } from 'recoil';
+import DatePicker from 'react-native-date-picker';
 import ButtonComponent from '../components/design/ButtonComponent';
 import ModalComponent from '../components/modal/ModalComponent';
+import { getFormattedDate } from '../utils/formattedDate';
+import { userState } from '../state/atoms/userAtom';
+import {
+  handleGettingCode,
+  handlePostingCode,
+} from '../api/connectCode/connectCode';
 
 const ConnectPartner = () => {
   const [visible, setModalVisible] = useState(false);
   const [inviteCode, setInviteCode] = useState('');
+  const [validCheckError, setValidCheckError] = useState(false);
   const [error, setError] = useState(false);
-  const [disabled, setDisabled] = useState(true);
+  const [date, setDate] = useState<Date | ''>('');
+  const [formattedDate, setFormattedDate] = useState<string>('');
+  const [open, setOpen] = useState(false);
+  const [myCode, setMyCode] = useState('');
+  const userInfo = useRecoilValue(userState);
+  const { memberId } = userInfo;
 
   const reg = /^(?=.*[a-zA-Z])(?=.*[0-9]).{6}$/;
   const validCheck = () => {
     if (reg.test(inviteCode)) {
+      setValidCheckError(false);
+      setError(false);
+    } else if (inviteCode.length === 0) {
+      setValidCheckError(false);
       setError(false);
-      setDisabled(false);
     } else {
-      setError(true);
-      setDisabled(true);
+      setValidCheckError(true);
     }
   };
```

Comment:
```suggestion
  const [errorText, setErrorText] = useState('');

  const validCheck = (code: string) => {
    const pattern = /^(?=.*[a-zA-Z])(?=.*[0-9]).{6}$/;

    if (code.length === 0) {
      setErrorText('');
    } else if (!pattern.test(code)) {
      setErrorText('잘못된 코드 형식입니다. 6자리 숫자, 문자');
    } else {
      setErrorText('');
    }
  };
  
// 버튼 비활성화
const disableButton = formattedDate.length === 0 || inviteCode.length === 0 || errorText;

// 일치하지 않는 코드 
 .catch((err) => {
          setErrorText('존재하지 않는 코드입니다.');
          console.log(err.response.data);

// error UI 
{errorText && <Text style={styles.errorText}>{errorText}</Text>}

```
이렇게 코드 수정해보시면 어떨까요? :)